### PR TITLE
Restaurant owner can update menu

### PR DESCRIPTION
**Pull request by GergKllai1**
- Adds create, update and delete method for restaurant owner
- Makes these functions available only for the restaurant owner in the index page
- Tests every method, both happy and sad paths

PT:
https://www.pivotaltracker.com/n/projects/2227327

screen shot 2018-11-29 at 16 50 01
screen shot 2018-11-29 at 16 50 06
screen shot 2018-11-29 at 16 50 11

### CODE REVIEW

File: features/restaurant_owner_can_update_menu.feature [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    I would like to be able to make updates to my menu
+
+    Background: 
+        Given the following dishes exist
+        | name        | description | price | category |
+        | Pizza       | Songoku     | 100   | Main     |
+        And I am logged in
+
+    Scenario: If restaruant owner fills in the update form correctly, he can update the menu
+        Given the following category exists
```
Comment (magnus-thor) [inline features/restaurant_owner_can_update_menu.feature:17]: Any reason for having this table within the scenario?

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -12,6 +18,11 @@
     click_on link
 end
 
+When("I choose {string} from {string}") do |option, select_box|
+    select_box = 'dish[category_id]'
```
Comment (magnus-thor) [inline features/step_definitions/basic_steps.rb]: You pass data into this step and then override it, either use the data you pass in or change the step.

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -26,20 +37,17 @@
 
 Given("I am logged in") do
     steps %{
-        Given I fill in 'Email' field with 'real@email.com'
+        Given the following user exists
+        | name | address      | phone   | email          | password | password_confirmation | restaurant_owner |
+        | Greg | 123 drive st.| 7654321 | really@email.com | password | password              | true            |
+        And I visit the page
+        And I click 'Log in'
+        Given I fill in 'Email' field with 'really@email.com'
```
Comment (magnus-thor) [inline features/step_definitions/basic_steps.rb]: Use a warden helper for this.

**Review by magnus-thor — CHANGES_REQUESTED** [review]

File: features/restaurant_owner_can_update_menu.feature (hunk 1 above)
Comment (GergKllai1) [reply features/restaurant_owner_can_update_menu.feature:17]: Yes, I want to have another category available to be able to change it.

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (GergKllai1) [reply features/step_definitions/basic_steps.rb]: Done

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (GergKllai1) [reply features/step_definitions/basic_steps.rb]: Done

**Review by magnus-thor — APPROVED** [review]

**Pull request merged**